test/e2e/shardedredisbackup_suite_test.go

```diff
@@ -0,0 +1,272 @@
+)
+
+const (
+	awsCredentials = "aws-credentials"
+	sshPrivateKey  = "redis-backup-ssh-private-key"
+	bucketName     = "my-bucket"
+	backupsPath    = "backups"
+)
+
+var _ = Describe("shardedredisbackup e2e suite", func() {
+	var ns string
+	var shards []saasv1alpha1.RedisShard
+	var sentinel saasv1alpha1.Sentinel
+	var backup saasv1alpha1.ShardedRedisBackup
+
+	BeforeEach(func() {
+		// Create a namespace for each block
+		ns = "test-ns-" + nameGenerator.Generate()
+
+		// Add any setup steps that needs to be executed before each test
+		testNamespace := &corev1.Namespace{
+			TypeMeta:   metav1.TypeMeta{APIVersion: "v1", Kind: "Namespace"},
+			ObjectMeta: metav1.ObjectMeta{Name: ns},
+		}
+
+		err := k8sClient.Create(context.Background(), testNamespace)
+		Expect(err).ToNot(HaveOccurred())
+
+		n := &corev1.Namespace{}
+		Eventually(func() error {
+			return k8sClient.Get(context.Background(), types.NamespacedName{Name: ns}, n)
+		}, timeout, poll).ShouldNot(HaveOccurred())
+
+		// create redis shards
+		shards = []saasv1alpha1.RedisShard{
+			{
+				ObjectMeta: metav1.ObjectMeta{Name: "rs0", Namespace: ns},
+				Spec: saasv1alpha1.RedisShardSpec{
+					MasterIndex: pointer.Int32(0),
+					SlaveCount:  pointer.Int32(2),
+					Command:     util.Pointer("/entrypoint.sh"),
+					Image: &saasv1alpha1.ImageSpec{
+						Name: util.Pointer("redis-with-ssh"),
+						Tag:  util.Pointer("4.0.11-alpine"),
+					},
+				},
+			},
+			// {
+			// 	ObjectMeta: metav1.ObjectMeta{Name: "rs1", Namespace: ns},
+			// 	Spec:       saasv1alpha1.RedisShardSpec{MasterIndex: pointer.Int32(2), SlaveCount: pointer.Int32(2)},
+			// },
+		}
+
+		for i, shard := range shards {
+			err = k8sClient.Create(context.Background(), &shard)
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() error {
+				err := k8sClient.Get(context.Background(), types.NamespacedName{Name: shard.GetName(), Namespace: ns}, &shard)
+				if err != nil {
+					return err
+				}
+				if shard.Status.ShardNodes != nil && shard.Status.ShardNodes.Master != nil {
+					// store the resource for later use
+					shards[i] = shard
+					GinkgoWriter.Printf("[debug] Shard %s topology: %+v\n", shard.GetName(), *shard.Status.ShardNodes)
+					return nil
+				} else {
+					return fmt.Errorf("RedisShard %s not ready", shard.ObjectMeta.Name)
+				}
+
+			}, timeout, poll).ShouldNot(HaveOccurred())
+		}
+
+		// create sentinel
+		sentinel = saasv1alpha1.Sentinel{
+			ObjectMeta: metav1.ObjectMeta{Name: "sentinel", Namespace: ns},
+			Spec: saasv1alpha1.SentinelSpec{
+				Replicas: util.Pointer(int32(1)),
+				Config: &saasv1alpha1.SentinelConfig{
+					MonitoredShards: map[string][]string{
+						shards[0].GetName(): {
+							"redis://" + shards[0].Status.ShardNodes.GetHostPortByPodIndex(0),
+							"redis://" + shards[0].Status.ShardNodes.GetHostPortByPodIndex(1),
+							"redis://" + shards[0].Status.ShardNodes.GetHostPortByPodIndex(2),
+						},
+						// shards[1].GetName(): {
```

Comment:
forgotten comment?

Reply:
oh, I actually intended to uncomment this to make the test use 2 shrads... but then I forgot :sweat_smile:

Reply:
lines deleted